src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -241,6 +241,23 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         });
     }
 
+    @Issue("JENKINS-34256")
+    @Test public void quietDownThenCancelQuietDown() {
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("semaphore 'wait'; echo 'I am done'", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/1", b);
+            r.jenkins.doQuietDown(true, 0);
+            SemaphoreStep.success("wait/1", null);
+            ((CpsFlowExecution) b.getExecution()).waitForSuspension();
```

Comment:
I would avoid using `waitForSuspension`. Really it should be deprecated—too internal. Would suffice to just sleep for a second here, I think.

Reply:
In https://github.com/jenkinsci/workflow-cps-plugin/pull/68 you switched from using `Thread.sleep(1000)` to `CpsFlowExecution.waitForSuspension` in `CpsFlowExecutionTest.quietDown` to try to fix some flaky tests, so I'm a bit hesitant to do the reverse. I'm happy to add a disclaimer to `CpsFlowExecution.waitForSuspension` that the method should not be used unless what you are testing involves some implementation detail of `CpsThreadGroup.scheduleRun`, if you think that would be helpful.

Reply:
#68 was a different case: the test would fail if the sleep were too short. Here, at worst, the test could _pass_ if the code were buggy.

Reply:
Ok, e026365 switches to using `waitForMessage` instead of a `waitForSuspension`. I think that is ok because the message is printed from inside of `scheduleRun`, but it might be flaky in some cases. I need to think about it some more.